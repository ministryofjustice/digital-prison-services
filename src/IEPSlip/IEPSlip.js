import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { H2 } from '@govuk-react/heading'
import { BORDER_COLOUR } from 'govuk-colours'
import { spacing } from '@govuk-react/lib'
import moment from 'moment'

import { properCaseName, pascalToString, forenameToInitial } from '../utils'
import { Location } from '../Location'
import { userType } from '../types'

const IEPSlipDetails = styled.div`
  display: none;
  border-bottom: 1px dashed ${BORDER_COLOUR};

  @media print {
    display: block;
  }
`

const Row = styled.div`
  display: flex;
  border-bottom: ${props => (props.noBorder ? 'none' : `1px solid ${BORDER_COLOUR}`)};
  margin-bottom: ${spacing.simple(3)}px;
  padding-bottom: ${spacing.simple(3)}px;
`

const Column = styled.div`
  flex: 1 0 0;
  padding-right: ${spacing.simple(3)}px;
`

const Value = styled.span`
  font-weight: bold;
  display: ${props => (props.inline ? 'inline' : 'block')};
`

const IEPSlip = ({ offender, iepValues, raisedDate, activityName, user }) => {
  const { absentReason, comments } = iepValues

  return (
    <IEPSlipDetails>
      <Row>
        <Column>
          <H2 size="MEDIUM" mb={0}>
            IEP Warning
          </H2>
        </Column>
        <Column>
          Date and time: <Value inline>{raisedDate.format('D MMMM YYYY, h:mm a')}</Value>
        </Column>
      </Row>
      <Row>
        <Column>
          Name:{' '}
          <Value>
            {properCaseName(offender.firstName)} {properCaseName(offender.lastName)}
          </Value>
        </Column>
        <Column>
          Number: <Value>{offender.offenderNo}</Value>
        </Column>
        <Column>
          Location: <Value>{<Location location={offender.cellLocation} agencyId={user.activeCaseLoadId} />}</Value>
        </Column>
        <Column>
          Activity or area: <Value>{activityName}</Value>
        </Column>
      </Row>

      <Row>
        <Column>
          Reason:
          <Value>
            {pascalToString(absentReason)}: {comments}
          </Value>
        </Column>
      </Row>
      <Row noBorder>
        <Column>
          Raised by: <Value inline>{forenameToInitial(user.name)}</Value>
        </Column>
        <Column>
          Date: <Value inline>{raisedDate.format('D MMMM YYYY')}</Value>
        </Column>
      </Row>
      <Row noBorder>
        <Column>Issued by:</Column>
        <Column>
          Date: <Value inline>{raisedDate.format('D MMMM YYYY')}</Value>
        </Column>
      </Row>
    </IEPSlipDetails>
  )
}

IEPSlip.propTypes = {
  user: userType.isRequired,
  raisedDate: PropTypes.instanceOf(moment),
  activityName: PropTypes.string.isRequired,
  offender: PropTypes.shape({}).isRequired,
  iepValues: PropTypes.shape({
    pay: PropTypes.string,
    absentReason: PropTypes.string,
    comments: PropTypes.string,
  }).isRequired,
}

IEPSlip.defaultProps = {
  raisedDate: moment(),
}

export default IEPSlip
