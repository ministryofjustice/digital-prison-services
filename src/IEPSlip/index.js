import IEPSlip from './IEPSlip'

export default IEPSlip
