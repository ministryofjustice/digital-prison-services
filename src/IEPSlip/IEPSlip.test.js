import React from 'react'
import moment from 'moment'
import { shallow } from 'enzyme'
import IEPSlip from '.'

const props = {
  offender: {
    firstName: 'Test',
    lastName: 'Offender',
    offenderNo: 'ABC123',
    cellLocation: 'LEI-123',
  },
  iepValues: {
    pay: 'false',
    absentReason: 'Refused',
    comments: 'Refused to attend',
  },
  activityName: 'Test activity',
  user: {
    activeCaseLoadId: 'LEI',
    name: 'Test User',
  },
  raisedDate: moment('2019-01-01'),
}
describe('<IEPSlip />', () => {
  it('matches the default snapshot', () => {
    const wrapper = shallow(<IEPSlip {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
