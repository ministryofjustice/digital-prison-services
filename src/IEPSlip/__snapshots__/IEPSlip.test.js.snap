// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IEPSlip /> matches the default snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <H2
        mb={0}
        size="MEDIUM"
      >
        IEP Warning
      </H2>
    </styled.div>
    <styled.div>
      Date and time: 
      <styled.span
        inline={true}
      >
        1 January 2019, 12:00 am
      </styled.span>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      Name:
       
      <styled.span>
        Test
         
        Offender
      </styled.span>
    </styled.div>
    <styled.div>
      Number: 
      <styled.span>
        ABC123
      </styled.span>
    </styled.div>
    <styled.div>
      Location: 
      <styled.span>
        <Location
          agencyId="LEI"
          location="LEI-123"
        />
      </styled.span>
    </styled.div>
    <styled.div>
      Activity or area: 
      <styled.span>
        Test activity
      </styled.span>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      Reason:
      <styled.span>
        Refused
        : 
        Refused to attend
      </styled.span>
    </styled.div>
  </styled.div>
  <styled.div
    noBorder={true}
  >
    <styled.div>
      Raised by: 
      <styled.span
        inline={true}
      >
        T User
      </styled.span>
    </styled.div>
    <styled.div>
      Date: 
      <styled.span
        inline={true}
      >
        1 January 2019
      </styled.span>
    </styled.div>
  </styled.div>
  <styled.div
    noBorder={true}
  >
    <styled.div>
      Issued by:
    </styled.div>
    <styled.div>
      Date: 
      <styled.span
        inline={true}
      >
        1 January 2019
      </styled.span>
    </styled.div>
  </styled.div>
</styled.div>
`;
